azerothcore/azerothcore-wotlk
# Adjust Scholomance Instructor Malicia abilities

## Changes Proposed:
##### Instructor Malicia added missing spells and adjusted present ones:
* Changed Corruption spell id to 18376
* Changed Flash Heal spell id to ~27608~ 17843
* Changed Renew spell id to 8362
* Added Heal with spell id ~22167~ 15586
* Added Slow with spell id ~6146~ 13747

## Issues Addressed:
https://github.com/azerothcore/azerothcore-wotlk/issues/4520
- Closes 4. Instructor malicia goes oom super fast, her corruptions cost 1k mana.(out of 7k)



## SOURCE:


##### Corruption:
According to https://youtu.be/2rj_ZQdNyRk?t=834 and https://youtu.be/s-2GXud7CJI?t=1589.
Corruption should be an AoE spell, have a cast time and provides:
```
around 220 damage per tick
417 mana cost
```
Currently Corruption is not an AoE spell and doesn't have a cast time at the moment provides:
```
around 111 damage per tick
1022 mana cost
```
##### Flash heal:
According to https://youtu.be/nWM5CGoHVZg?t=2860.
Flash Heal should provide:
```
around 810 healing
521 mana cost
```
Currently Flash heal provides:
```
around 844 - 943 healing
1314 mana cost
```
##### Renew:
According to https://youtu.be/3xOWaxQqIYE?t=1076.
Renew should provide:
```
around 347 healing per tick
300 mana cost
```
Currently Renew provides:
```
around 162 healing per tick
1241 mana cost
```
##### Heal:
According to https://youtu.be/3xOWaxQqIYE?t=1040
Below 40% Malicia casts Heal:
```
3680 healing
896 mana cost
```
Currently doesn't cast Heal at all.
##### Slow:
According to this video https://youtu.be/s-2GXud7CJI?t=1593 Malicia should cast slow on multiple targets.

Currently she doesn't cast Slow at all.
## Tests Performed:


- Corruption:
  - Currently Corruption is not an AoE spell and doesn't have a cast time at the moment provides around 111 damage per tick
for 1022 mana cost.
  - Tested in game on linux/docker setup from windows client.
  - With the changes Malicia casts AoE Corruption with spell id 18376 and cast time of 2 seconds for:
  ```
  around 250 damage per tick on (0 resistance target)
  417 mana cost
  ```
- Flash Heal:
  - Currently Flash heal provides around 844 - 943 healing 1314 mana cost.
  - Tested in game on linux/docker setup from windows client.
  - With the changes Malicia casts Flash Heal with spell id 27608 for around 857 healing and 380 mana cost.
- Renew:
  - Currently casts Renew provides around 162 healing per tick and 1241 mana cost..
  - Tested in game on linux/docker setup from windows client.
  - With the changes Malicia casts Renew with spell id 8362 for around 347 healing per tick with no mana cost.(I wasn't able to find a better spell that fits our needs)
- Heal:
  - Currently doesn't cast Heal at all.
  - Tested in game on linux/docker setup from windows client.
  - With the changes casts Heal with spell id 22167 for 4087 healing and 896 mana cost.
- Slow:
  - Currently doesn't cast Slow at all.
  - Tested in game on linux/docker setup from windows client.
  - With the changes Malicia casts Slow 6146 on three party members.(Victim, HOSTILE_SECOND_AGGRO, HOSTILE_LAST_AGGRO)
Screenshots after the adjustments:
![Screenshot_20210410_134616](https://user-images.githubusercontent.com/10816895/114300014-ddc7e800-9ac6-11eb-9fb5-82250493c8c0.png)
![Screenshot_20210410_134727](https://user-images.githubusercontent.com/10816895/114300019-e15b6f00-9ac6-11eb-9b49-8692db89c844.png)
![Screenshot_20210410_134749](https://user-images.githubusercontent.com/10816895/114300022-e4eef600-9ac6-11eb-89dd-5717621a74f2.png)
![Screenshot_20210410_134801](https://user-images.githubusercontent.com/10816895/114300025-e7e9e680-9ac6-11eb-89eb-0995f52205c6.png)

## How to Test the Changes:

1. Go to Scholomance and start encounter with Instructor Malicia.
2. Test if the current state is as described above.
3. Apply the DB query.
4. Go to Scholomance and start encounter with Instructor Malicia.
5. Check if the above spell descriptions are correct.

## Known Issues and TODO List:


## Target Branch(es):







 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
